docs/index.md

```diff
@@ -58,3 +58,60 @@ another route, based on the changes that the filters made to the
 request.
 
 ![Skipper's request and response processing ](/img/req-and-resp-processing.svg)
+
+#### Routing mechanism
+
+The routing executes the following steps in the typical case:
+
+1. Select the best fitting route by matching the request against the
+   predicates. When no route found, respond with the default status code
+   (the default of the default is 404).
+
+2. Execute the filters defined in the route in normal order on the
+   request. The filters may or may not alter the request.
+
+3. Forward the request to the backend defined by the route and receive
+   a response.
+
+4. Execute the filters defined in the route in reverse order on the
+   response. The filters may or may not alter the response.
+
+5. Respond to the incoming request with the resulting response.
+
+#### Route matching
+
+Skipper can handle a relatively large number of routes acceptable
+performance, while being able to use any attribute of the incoming HTTP
+requests to distinguis between them. In order to be able to do so, the
+path matching predicates (Path() and PathSubtree() but not PathRegexp())
```

Comment:
nitpicking: Probably code formatting for the filter names would be better here and below